integration_tests/commands/resp/set_data_cmd_test.go

```diff
@@ -0,0 +1,164 @@
+		},
+		{
+			name:       "SADD & CARD with non existing key",
+			cmd:        []string{"SADD foo bar", "SADD foo baz", "SCARD bar"},
+			expected:   []interface{}{int64(1), int64(1), int64(0)},
+			assertType: []string{"equal", "equal", "equal"},
+			delay:      []time.Duration{0, 0, 0},
+		},
+		{
+			name:       "SADD & SCARD with wrong key type",
+			cmd:        []string{"SET foo bar", "SCARD foo"},
+			expected:   []interface{}{"OK", "WRONGTYPE Operation against a key holding the wrong kind of value"},
+			assertType: []string{"equal", "equal"},
+			delay:      []time.Duration{0, 0},
+		},
+		// SMEMBERS
+		{
+			name:       "SADD & SMEMBERS",
+			cmd:        []string{"SADD foo bar", "SADD foo baz", "SMEMBERS foo"},
+			expected:   []interface{}{int64(1), int64(1), []any{"bar", "baz"}},
+			assertType: []string{"equal", "equal", "equal"},
+			delay:      []time.Duration{0, 0, 0},
+		},
+		{
+			name:       "SADD & SMEMBERS with non existing key",
+			cmd:        []string{"SMEMBERS foo"},
+			expected:   []interface{}{[]any{}},
+			assertType: []string{"equal"},
+			delay:      []time.Duration{0},
+		},
+		{
+			name:       "SADD & SMEMBERS with wrong key type",
+			cmd:        []string{"SET foo bar", "SMEMBERS foo"},
+			expected:   []interface{}{"OK", "WRONGTYPE Operation against a key holding the wrong kind of value"},
+			assertType: []string{"equal", "equal"},
+			delay:      []time.Duration{0, 0},
+		},
+		// SREM
+		{
+			name:       "SADD & SREM",
+			cmd:        []string{"SADD foo bar", "SADD foo baz", "SREM foo bar", "SMEMBERS foo"},
+			expected:   []interface{}{int64(1), int64(1), int64(1), []any{"baz"}},
+			assertType: []string{"equal", "equal", "equal", "equal"},
+			delay:      []time.Duration{0, 0, 0, 0},
+		},
+		{
+			name:       "SADD & SREM with non existing key",
+			cmd:        []string{"SREM foo bar"},
+			expected:   []interface{}{int64(0)},
+			assertType: []string{"equal"},
+			delay:      []time.Duration{0},
+		},
+		{
+			name:       "SADD & SREM with wrong key type",
+			cmd:        []string{"SET foo bar", "SREM foo bar"},
+			expected:   []interface{}{"OK", "WRONGTYPE Operation against a key holding the wrong kind of value"},
+			assertType: []string{"equal", "equal"},
+			delay:      []time.Duration{0, 0},
+		},
+		{
+			name:       "SADD & SREM with non existing value",
+			cmd:        []string{"SADD foo bar baz bax", "SMEMBERS foo", "SREM foo bat", "SMEMBERS foo"},
+			expected:   []interface{}{int64(3), []any{"bar", "baz", "bax"}, int64(0), []any{"bar", "baz", "bax"}},
+			assertType: []string{"equal", "equal", "equal", "equal"},
+			delay:      []time.Duration{0, 0, 0, 0},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			FireCommand(conn, "DEL foo")
+			FireCommand(conn, "DEL foo2")
+			for i, cmd := range tc.cmd {
+				if tc.delay[i] > 0 {
+					time.Sleep(tc.delay[i])
+				}
+				result := FireCommand(conn, cmd)
+				CustomDeepEqual(t, result, tc.expected[i])
+			}
```

Comment:
Please add a final cleanup before exiting the test run

Reply:
@apoorvyadav1111 If this is not a hard blocker for the merge can we take it in a separate issue?

Reply:
not a hard blocker as I think foo, foo2 are unique keys